@navikt/core/tokens/darkside/figma/plugin/AkselVariablesInterface.ts

```diff
@@ -0,0 +1,258 @@
+import _config from "../../../figma-config.json";
+import { FigmaConfigEntry, FigmaTokenConfig } from "../figma-config.types";
+import { FigmaPluginInterface } from "./FigmaPluginInterface";
+
+type ScopedFigmaTokenConfig = Omit<FigmaTokenConfig, "version" | "timestamp">;
+
+export class AkselVariablesInterface {
+  private Figma: FigmaPluginInterface;
+  private config: FigmaTokenConfig;
+  private meta: Pick<FigmaTokenConfig, "version" | "timestamp">;
+  private remoteConfigURL =
+    "https://cdn.nav.no/designsystem/@navikt/tokens/figma-config.json";
+
+  constructor() {
+    const config = _config as FigmaTokenConfig;
+    this.config = config;
+    this.meta = { timestamp: config.timestamp, version: config.version };
+    this.Figma = new FigmaPluginInterface();
+  }
+
+  /**
+   * Because of "dynamic-page" loading in Figma,
+   * we need to initialize the plugin trough async methods.
+   * This method should be called before any other methods.
+   */
+  async init(): Promise<void> {
+    await this.Figma.init();
+  }
+
+  async useRemoteConfig(): Promise<void> {
+    const newConfig = await fetch(this.remoteConfigURL)
+      .then((res) => res.json())
+      .catch((err) => {
+        console.error(err);
+        throw new Error("Error fetching config from CDN 😱");
+      });
+    this.meta = { timestamp: newConfig.timestamp, version: newConfig.version };
+
+    this.config = newConfig;
+  }
+
+  updateVariables(): void {
+    this.updateGlobalColorCollection(this.config.colors.light.global);
+    this.updateGlobalColorCollection(this.config.colors.dark.global);
+    this.updateSemanticColorCollection(this.config.colors);
+    this.updateScaleCollection(this.config.radius);
+    this.updateScaleCollection(this.config.spacing);
+    console.info("Updated variables!");
+  }
+
+  private updateGlobalColorCollection(
+    globalColorTheme: FigmaConfigEntry,
+  ): void {
+    let collection = this.Figma.getCollection(globalColorTheme.name);
+
+    if (!collection) {
+      collection = this.Figma.createCollection(globalColorTheme.name);
+    }
+
+    /**
+     * Correctly sorts the token-scale for global colors
+     * 000 - 100 - 200... etc.
+     */
+    const sortedTokens = globalColorTheme.tokens.sort((a, b) => {
+      const getLastNumber = (name: string) => {
+        const matches = name.match(/\d+/g);
+        return matches ? parseInt(matches[matches.length - 1], 10) : 0;
+      };
+      return getLastNumber(a.name) - getLastNumber(b.name);
+    });
```

Comment:
Probably micro-optimization, but it just feels wrong to define a function inside sort 😅
```suggestion
    const getLastNumber = (name: string) => {
      const matches = name.match(/\d+/g);
      return matches ? parseInt(matches[matches.length - 1], 10) : 0;
    };
    const sortedTokens = globalColorTheme.tokens.sort(
      (a, b) => getLastNumber(a.name) - getLastNumber(b.name),
    );
```